test/org/zaproxy/zap/authentication/UsernamePasswordAuthenticationCredentialsTest.java

```diff
@@ -0,0 +1,115 @@
+import static org.junit.Assert.*;
+
+
+public class UsernamePasswordAuthenticationCredentialsTest {
+
+    private UsernamePasswordAuthenticationCredentials usernamePasswordAuthenticationCredentials;
+    private ApiResponseSet apiResponseSet;
+    private String username = "username";
+    private String password = "password";
+
+    @Before
+    public void setUp() {
+        this.usernamePasswordAuthenticationCredentials = new UsernamePasswordAuthenticationCredentials(username, password);
+    }
+
+    @Test
+    public void shouldNotUsernameBeEmptyOrNull() throws Exception {
+        // Given
+        String username = usernamePasswordAuthenticationCredentials.getUsername();
+
+        // When/Then
+        assertThat(username, not(isEmptyOrNullString()));
+    }
+
+    @Test
+    public void shouldNotPasswordBeEmptyOrNull() throws Exception {
+        // Given
+        String password = usernamePasswordAuthenticationCredentials.getPassword();
+
+        // When/Then
+        assertThat(password, not(isEmptyOrNullString()));
+    }
+
+    @Test
+    public void shouldReturnTureIfIsConfigured() throws Exception {
+        // Given
+        boolean isConfigured = usernamePasswordAuthenticationCredentials.isConfigured();
+
+        // When/Then
+        assertThat(isConfigured, is(true));
+    }
+
+    @Test
+    public void shouldEncodeReturnNullCredentialsIfUsernameIsNull() throws Exception {
+        // Given
+        username = null;
+        password = "something";
+        String parentStringSeparator = "|";
+        usernamePasswordAuthenticationCredentials = new UsernamePasswordAuthenticationCredentials(username, password);
+
+        // When
+        String encodedResult = usernamePasswordAuthenticationCredentials.encode(parentStringSeparator);
+
+        // Then
+        assertThat(encodedResult, is("AA=="));
+    }
+
+    @Test(expected = AssertionError.class)
+    public void shouldParentStringSeparatorNotBeFieldSeparatorWhenEncode() throws Exception {
+        // Given
+        String parentStringSeparator = "~";
+        usernamePasswordAuthenticationCredentials = new UsernamePasswordAuthenticationCredentials(username, password);
+
+        // When
+        usernamePasswordAuthenticationCredentials.encode(parentStringSeparator);
+
+        // Then = AssertionError
+    }
+
+    @Test
+    public void shouldEncodeUsernameAndPasswordWithTheRightFieldSeparator() throws Exception {
+        // Given
+        String parentStringSeparator = "|";
+
+        // When
+        String encodedUsernamePassword = usernamePasswordAuthenticationCredentials.encode(parentStringSeparator);
+
+        // Then
+        assertThat(encodedUsernamePassword, not(isEmptyOrNullString()));
+        assertThat(encodedUsernamePassword, allOf(containsString("dXNlcm5hbWU="),
+                                                    containsString("cGFzc3dvcmQ="),
+                                                    containsString("~")));
```

Comment:
I'm going to make it broader..
By making a list of non-field-separator samples, randomly picking an element from it, and feeding it to the `encode()` method. 
I will refactor the method name to reflect this